python/ray/experimental/client/common.py

```diff
@@ -41,59 +46,143 @@ def remote(self, *args, **kwargs):
         return ray.call_remote(self, ray_client_pb2.ClientTask.FUNCTION, *args,
                                **kwargs)
 
+    def _get_ray_remote_impl(self):
+        if self._raylet_remote is None:
+            self._raylet_remote = ray.remote(self._func)
+        return self._raylet_remote
+
     def __repr__(self):
         return "ClientRemoteFunc(%s, %s)" % (self._name, self.id)
 
+    def _prepare_client_task(self) -> ray_client_pb2.ClientTask:
+        if self._ref is None:
+            self._ref = ray.put(self._func)
```

Comment:
Can you document the attributes of this class with pydoc, and comment as to why this put needs to be lazy?

Reply:
the comment is in __init__ (because that's the place where it would otherwise be created) but otherwise, done.